File: backend/src/main/java/com/example/madrid/madrid/Establishment.java
package com.example.madrid.madrid;

import java.math.BigDecimal;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;


@Entity
public class Establishment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id")
    private Long id;
    private String name;
    private String description;
    private BigDecimal latitude;
    private BigDecimal longitude;
    private String website;
    private String telephone;
    @OneToOne
    @JoinColumn(name = "userName", referencedColumnName = "userName")
    private User submitter;

    // Getters and setters
    
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public BigDecimal getLatitude() {
        return latitude;
    }

    public void setLatitude(BigDecimal latitude) {
        this.latitude = latitude;
    }

    public BigDecimal getLongitude() {
        return longitude;
    }

    public void setLongitude(BigDecimal longitude) {
        this.longitude = longitude;
    }

    public String getWebsite() {
        return website;
    }

    public void setWebsite(String website) {
        this.website = website;
    }

    public String getTelephone() {
        return telephone;
    }

    public void setTelephone(String telephone) {
        this.telephone = telephone;
    }

    public User getSubmitter() {
        return submitter;
    }

    public void setSubmitter(User submitter) {
        this.submitter = submitter;
    }
}

File: frontend/src/reducers/establishment.js
// reducers/establishments.js
const initialState = {
    name: null,
    description: null,
    latitude: null,
    longitude: null,
    website: null,
    telephone: null,
    submitter: null
  };
  
  const establishmentReducer = (state = initialState, action) => {
    switch (action.type) {
      case 'ADD':
        return { ...state,
            price: state.price, 
            description: state.description, 
            latitude: state.latitude, 
            longitude: state.longitude,
            website: state.website,
            telephone: state.telephone,
            submitter: state.submitter
            };
      case 'CLEAR':
        return {
            name: null,
            description: null,
            latitude: null,
            longitude: null,
            website: null,
            telephone: null,
            submitter: null
        }
      default:
        return state;
    }
  };
  
  export default establishmentReducer;
  